### DISCUSSION TITLE
Find First and Last Position of Element in Sorted Array

### DISCUSSION THREAD

**Discussion by nfay55** [community/community#54171] [Programming Help] [answered]
Given an array of integers `nums` sorted in non-decreasing order, find the starting and ending position of a given `target` value.

If `target` is not found in the array, return `[-1, -1]`.

You must write an algorithm with `O(log n)` runtime complexity.

Example 1:

```
Input: nums = [5,7,7,8,8,10], target = 8
Output: [3,4]
```

Example 2:

```
Input: nums = [5,7,7,8,8,10], target = 6
Output: [-1,-1]
```

Constraints:

1. 0 <= nums.length <= 105
2. -109 <= nums[i] <= 109
3. nums is a non-decreasing array.
4. -109 <= target <= 109

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by FEsoftdev** [comment, marked as answer]
There is a Javascript solution for the above problem.

```
var searchRange = function(nums, target) {
    let left = 0, right = nums.length - 1;
    let leftIndex = -1, rightIndex = -1;
    
    while (left <= right) {
        let mid = Math.floor((left + right) / 2);
        
        if (nums[mid] >= target) {
            right = mid - 1;
        } else {
            left = mid + 1;
        }
        
        if (nums[mid] === target) {
            leftIndex = mid;
        }
    }
    
    left = 0, right = nums.length - 1;
    
    while (left <= right) {
        let mid = Math.floor((left + right) / 2);
        
        if (nums[mid] <= target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
        
        if (nums[mid] === target) {
            rightIndex = mid;
        }
    }
    
    return [leftIndex, rightIndex];
};
```

Please check and let me know how it works!

**Reply by nfay55** [reply]
Thanks, it works perfectly.
